packages/plugin-ext/src/hosted/browser/hosted-plugin-watcher.ts

```diff
@@ -30,7 +30,9 @@ export class HostedPluginWatcher {
     getHostedPluginClient(): HostedPluginClient {
         const messageEmitter = this.onPostMessage;
         const logEmitter = this.onLogMessage;
+        const clientId = Math.floor(Math.random() * 1000000) + 1;
         return {
+            getClientId: () => Promise.resolve(clientId),
```

Comment:
Should not server create client handles then? It knows everything about them.

Reply:
@akosyakov could you point me to some example where client handles are used?

Reply:
I mean server can give a client an id, since it knows all clients and can deduplicate them. Instead of relying on random numbers.

Reply:
You mean I should add to `HostedPluginClient` a method `setClientId` that the server populates instead of populating it in the client?

Reply:
ok. if i understood correctly then the code is changed now.